### PR TITLE
Updated tests and changed from default exports

### PR DESCRIPTION
**Pull request by rob3000**
* Changed tests to typescript.
* Removed default exports to named exports.
* Updated linting rules to allow smart-tabs

### CODE REVIEW

**Comment by wtrocki** [comment]
Awesome work! Thank you so much. I will review this soon

File: .eslintrc.js [hunk 1]
```diff
@@ -29,6 +30,7 @@ module.exports = {
       "error",
       "always"
     ],
-    "no-console": 0
+    "no-console": 0,
+    "no-mixed-spaces-and-tabs": ["error", "smart-tabs"]
```
Comment (wtrocki) [inline .eslintrc.js:34]: Typically entire org has the same linting rules
For node-sinek it is:
https://github.com/nodefluent/node-sinek/blob/master/.eslintrc.js

I would preffer no smart-tabs tbh. 
Generally I see linting as separate issue/PR so we can get typescript there with still history or original changes and then relint the code based on rules that can be applied to every project - not just kafka-streams.

Linting will also obfuscate review and will make it hard to spot some issues with migration.

File: src/index.ts [hunk 2]
```diff
@@ -1,18 +1,7 @@
 "use strict";
```
Comment (wtrocki) [inline src/index.ts:1]: I used this for testing - it is recommended to have this when migration. I seen that some of those were removed. I will remove it everywhere later.

**Comment by wtrocki** [comment]
Amazing job but I would prefer linting rules (this completely separate concern) to not obfuscate typescript migration if possible. If you agree I will revert linting rules when mering this

**Comment by rob3000** [comment]
sounds good! Happy to tackle the linting rules after :)

**Pull request merged**